test/versioned/langchain/tools.tap.js

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const helper = require('../../lib/agent_helper')
+// load the assertSegments assertion
+require('../../lib/metrics_helper')
+const fs = require('fs')
+// have to read and not require because openai does not export the package.json
+const { version: pkgVersion } = JSON.parse(
+  fs.readFileSync(`${__dirname}/node_modules/@langchain/core/package.json`)
+)
+const { DESTINATIONS } = require('../../../lib/config/attribute-filter')
+const config = {
+  ai_monitoring: {
+    enabled: true
+  },
+  feature_flag: {
+    langchain_instrumentation: true
+  }
+}
+
+tap.test('Langchain instrumentation - tools', (t) => {
+  t.autoend()
+
+  t.before(() => {
+    t.context.agent = helper.instrumentMockedAgent(config)
+    const { WikipediaQueryRun } = require('@langchain/community/tools/wikipedia_query_run')
+    t.context.tool = new WikipediaQueryRun({
+      topKResults: 3,
+      maxDocContentLength: 4000
+    })
```

Comment:
If we are going to use a real community module, we should definitely mock the outgoing HTTP requests. I understand that versioned tests are a sort of integration test suite, but relying on external network resources usually ends in tears (recall https://github.com/newrelic/node-newrelic/pull/1937).

Reply:
yea I can do that

Reply:
actually i think I'm going to create a test tool